File: motor/modelo.py
"""Modelo de datos central del motor."""

from dataclasses import dataclass, field


@dataclass
class Chapter:
    """Un capítulo del libro: título + cuerpo HTML ya limpio."""

    titulo: str
    html_cuerpo: str


@dataclass
class Nota:
    """Nota al pie extraída del manuscrito."""

    num: int
    texto: str
    cap_num: int | None = None


@dataclass
class Contadores:
    """Estadísticas del libro para el dashboard de la UI."""

    capitulos: int
    notas: int
    imagenes: int
    separadores: int


@dataclass
class Resultado:
    """Salida completa del pipeline: capítulos, notas, estadísticas y avisos."""

    capitulos: list[Chapter]
    notas: list[Nota]
    contadores: Contadores
    avisos: list[str] = field(default_factory=list)

File: app_flutter/lib/motor/modelo.dart
class Chapter {
  String titulo;
  String htmlCuerpo;
  String htmlRaw;
  String? archivo;
  String? plantillaRuta;
  String? plantillaNombre;
  bool tituloEsImagen;
  String? numeroImagenTitulo;

  Chapter({
    required this.titulo,
    required this.htmlCuerpo,
    this.htmlRaw = "",
    this.archivo,
    this.plantillaRuta,
    this.plantillaNombre,
    this.tituloEsImagen = false,
    this.numeroImagenTitulo,
  });
}

class Nota {
  int num;
  String texto;
  int? capNum;
  String? capArchivo;

  Nota({
    required this.num,
    required this.texto,
    this.capNum,
    this.capArchivo,
  });
}

class Contadores {
  int capitulos;
  int notas;
  int imagenes;
  int separadores;

  Contadores({
    required this.capitulos,
    required this.notas,
    required this.imagenes,
    required this.separadores,
  });
}

class Resultado {
  List<Chapter> capitulos;
  List<Nota> notas;
  Contadores contadores;
  List<String> avisos;

  Resultado({
    required this.capitulos,
    required this.notas,
    required this.contadores,
    List<String>? avisos,
  }) : avisos = avisos ?? [];
}
